src/univariate/continuous/johnson.jl

```diff
@@ -0,0 +1,141 @@
+"""
+    Johnson(ξ, λ, γ, δ)
+
+The Johnson's ``S_U``-distribution with parameters ξ, λ, γ and δ is a transformation of the normal distribution:
+
+```math
+z \\sim \\gamma + \\delta \\sinh^{-1}\\Big(\\frac{x-\\xi}{\\lambda}\\Big),
+```
+
+where ``z \\sim \\mathcal{N}(0,1)``.
+
+```julia
+Johnson()           # Equivalent to Johnson(0.0, 1.0, 0.0, 1.0)
+Johnson(ξ, λ, γ, δ) # Johnson's S_U-distrubtion with shape parameters ξ, λ, γ and δ
+
+params(d)           # Get the parameters, i.e. (ξ, λ, γ, δ)
+shape(d)            # Get the shape parameter, i.e. ξ
+scale(d)            # Get the scale parameter, i.e. λ
+```
+
+External links
+
+* [Johnson's ``S_U``-distribution on Wikipedia](http://en.wikipedia.org/wiki/Johnson%27s_SU-distribution)
+"""
+struct Johnson{T<:Real} <: ContinuousUnivariateDistribution
+	ξ::T
+	λ::T
+	γ::T
+	δ::T
+	Johnson{T}(ξ, λ, γ, δ) where {T} = new{T}(ξ, λ, γ, δ)
+end
+
+function Johnson(ξ::T, λ::T, γ::T, δ::T; check_args::Bool=true) where {T<:Real}
+    @check_args Johnson (λ, λ ≥ zero(λ)) (δ, δ ≥ zero(δ))
+	return Johnson{T}(ξ, λ, γ, δ)
+end
+
+Johnson() = Johnson{Float64}(0.0, 1.0, 0.0, 1.0)
+Johnson(ξ::Real, λ::Real, γ::Real, δ::Real; check_args::Bool=true) = Johnson(promote(ξ, λ, γ, δ)...; check_args=check_args)
+Johnson(ξ::Integer, λ::Integer, γ::Integer, δ::Integer; check_args::Bool=true) = Johnson(float(ξ), float(λ), float(γ), float(δ); check_args=check_args)
+
+@distr_support Johnson -Inf Inf
+
+#### Conversions
+
+convert(::Type{Johnson{T}}, ξ::S, λ::S, γ::S, δ::S) where {T<:Real, S<:Real} = Johnson(T(ξ), T(λ), T(γ), T(δ))
+Base.convert(::Type{Johnson{T}}, d::Johnson) where {T<:Real} = Johnson{T}(T(d.ξ), T(d.λ), T(d.γ), T(d.δ))
+Base.convert(::Type{Johnson{T}}, d::Johnson{T}) where {T<:Real} = d
+
+#### Parameters
+
+shape(d::Johnson) = d.ξ
+scale(d::Johnson) = d.λ
+
+params(d::Johnson) = (d.ξ, d.λ, d.γ, d.δ)
+partype(d::Johnson{T}) where {T<:Real} = T
+
+#### Statistics
+
+function mean(d::Johnson)
+    a = exp(d.δ^-2)
+    r = d.γ/d.δ
+    d.ξ - d.λ * sqrt(a) * sinh(r)
+end
+function median(d::Johnson)
+    r = d.γ/d.δ
+    d.ξ + d.λ * sinh(-r)
+end
+function var(d::Johnson)
+    a = exp(d.δ^-2)
+    r = d.γ/d.δ
+    d.λ^2/2 * (a-1) * (a*cosh(2r)+1)
+end
+function skewness(d::Johnson)
+    a = exp(d.δ^-2)
+    r = d.γ/d.δ
+    - (d.λ^3 * sqrt(a) * (a-1)^2 * (a*(a+2)*sinh(3r)+3sinh(2r))) / 4var(d)^1.5
```

Comment:
Please avoid floating literal floating point numbers as often they will cause undesired promotions.

Reply:
I suppose `sqrt(var(d)^3)` can avoid that?

Reply:
You could use `3//2` rather than `1.5`

Reply:
You could use `sqrt(x) * x` which will be much faster and slightly more accurate than `x^(3/2)`.